sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1759,6 +1759,162 @@ def _df_getitem_unicode_idx_impl(self, idx):
     ty_checker.raise_exc(idx, expected_types, 'idx')
 
 
+def df_getitem_int_iloc_codegen(self, idx):
+    """
+    Example of generated implementation:
+        def _df_getitem_int_iloc_impl(self, idx):
+        if -1 < idx < len(self._dataframe.index):
+            data_0 = pandas.Series(self._dataframe._data[0])
+            result_0 = data_0.iat[idx]
+            data_1 = pandas.Series(self._dataframe._data[1])
+            result_1 = data_1.iat[idx]
+            return pandas.Series(data=[result_0, result_1], index=['A', 'B'], name=str(idx))
+        raise IndexingError('Index is out of bounds for axis')
+    """
+    func_lines = ['def _df_getitem_int_iloc_impl(self, idx):',
+                  '  if -1 < idx < len(self._dataframe.index):']
+    results = []
+    index = []
+    name = df_name_codegen_iloc(self)
+    for i, c in enumerate(self.columns):
+        result_c = f"result_{i}"
+        func_lines += [f"    data_{i} = pandas.Series(self._dataframe._data[{i}])",
+                       f"    {result_c} = data_{i}.iat[idx]"]
+        results.append(result_c)
+        index.append(c)
+    data = ', '.join(col for col in results)
+    func_lines += [f"    return pandas.Series(data=[{data}], index={index}, name=str({name}))",
```

Comment:
Shouldn't you join `index` as well as you joined `results`? Why not insert `name` directly without type casting?